File: openapi-client/zapier/models/MailBlockClickHouse.js
const utils = require('../utils/utils');

module.exports = {
    fields: (prefix = '', isInput = true, isArrayChild = false) => {
        const {keyPrefix, labelPrefix} = utils.buildKeyAndLabel(prefix, isInput, isArrayChild)
        return [
            {
                key: `${keyPrefix}date`,
                label: `The date the block event was recorded. - [${labelPrefix}date]`,
                required: true,
                type: 'string',
            },
            {
                key: `${keyPrefix}from`,
                label: `The SMTP envelope sender (`MAIL FROM`) address of the blocked message. Pass this value as `email` to `POST /mail/blocks/delete` to delist it. - [${labelPrefix}from]`,
                required: true,
                type: 'string',
            },
            {
                key: `${keyPrefix}subject`,
                label: `The `Subject` header of the blocked message. - [${labelPrefix}subject]`,
                required: true,
                type: 'string',
            },
            {
                key: `${keyPrefix}to`,
                label: `The serialized list of recipients of the blocked message. - [${labelPrefix}to]`,
                required: true,
                type: 'string',
            },
            {
                key: `${keyPrefix}messageId`,
                label: `The `Message-ID` header of the blocked message, or `null` if not present. - [${labelPrefix}messageId]`,
                type: 'string',
            },
        ]
    },
    mapping: (bundle, prefix = '') => {
        const {keyPrefix} = utils.buildKeyAndLabel(prefix)
        return {
            'date': bundle.inputData?.[`${keyPrefix}date`],
            'from': bundle.inputData?.[`${keyPrefix}from`],
            'subject': bundle.inputData?.[`${keyPrefix}subject`],
            'to': bundle.inputData?.[`${keyPrefix}to`],
            'messageId': bundle.inputData?.[`${keyPrefix}messageId`],
        }
    },
}

File: openapi-client/scalaz/src/main/scala/org/openapitools/client/api/MailBlockClickHouse.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime

import java.time.LocalDate

import MailBlockClickHouse._

case class MailBlockClickHouse (
  /* The date the block event was recorded. */
  date: LocalDate,
/* The SMTP envelope sender (`MAIL FROM`) address of the blocked message. Pass this value as `email` to `POST /mail/blocks/delete` to delist it. */
  from: String,
/* The `Subject` header of the blocked message. */
  subject: String,
/* The serialized list of recipients of the blocked message. */
  to: String,
/* The `Message-ID` header of the blocked message, or `null` if not present. */
  messageId: Option[String])

object MailBlockClickHouse {
  import DateTimeCodecs._

  implicit val MailBlockClickHouseCodecJson: CodecJson[MailBlockClickHouse] = CodecJson.derive[MailBlockClickHouse]
  implicit val MailBlockClickHouseDecoder: EntityDecoder[MailBlockClickHouse] = jsonOf[MailBlockClickHouse]
  implicit val MailBlockClickHouseEncoder: EntityEncoder[MailBlockClickHouse] = jsonEncoderOf[MailBlockClickHouse]
}
